File: fotoapparat/src/main/java/io/fotoapparat/view/CameraRenderer.java
package io.fotoapparat.view;

import io.fotoapparat.hardware.CameraDevice;
import io.fotoapparat.parameter.ScaleType;

/**
 * Renders the stream from {@link io.fotoapparat.Fotoapparat}.
 */
public interface CameraRenderer {

    /**
     * Sets the scale type of the preview to the renderer. This method will be called from camera
     * thread, so it is safe to perform blocking operations here.
     */
    void setScaleType(ScaleType scaleType);

    /**
     * Attaches renderer to camera, so that it will display the preview when camera is started. This
     * method will be called from camera thread, so it is safe to perform blocking operations here.
     */
    void attachCamera(CameraDevice camera);

}

File: fotoapparat/src/main/java/io/fotoapparat/view/CameraRenderer.kt
package io.fotoapparat.view

import io.fotoapparat.parameter.Resolution
import io.fotoapparat.parameter.ScaleType

/**
 * Renders the stream from [io.fotoapparat.Fotoapparat].
 */
interface CameraRenderer {

    /**
     * Sets the scale type of the preview to the renderer. This method will be called from camera
     * thread, so it is safe to perform blocking operations here.
     */
    fun setScaleType(scaleType: ScaleType)

    /**
     * Sets the resolution at which the view should display the preview.
     */
    fun setPreviewResolution(resolution: Resolution)

    /**
     * Returns the surface texture when available.
     */
    fun getPreview(): Preview

}
